pkg/sql/catalog/descs/schema.go

```diff
@@ -114,6 +114,15 @@ func (tc *Collection) getSchemaByID(
 	if sc, err := tc.virtual.getSchemaByID(
 		ctx, schemaID, flags.RequireMutable,
 	); sc != nil || err != nil {
+		if err != nil {
+			if errors.Is(err, catalog.ErrDescriptorNotFound) {
+				if flags.Required {
+					return nil, sqlerrors.NewUndefinedSchemaError(fmt.Sprintf("[%d]", schemaID))
+				}
+				return nil, nil
+			}
+			return nil, err
+		}
```

Comment:
uber nit: in order to reduce the number of levels of nesting, you could consider handling this in a `defer`, for example:
```
defer func() {
    if !errors.Is(err, catalog.ErrDescriptorNotFound) {
        return
    }
    if flags.Required {
        err = sqlerrors.NewUndefinedSchemaError(fmt.Sprintf("[%d]", schemaID))
    } else {
        err = nil
    }
}()